controllers/osbuildenvconfig_controller_test.go

```diff
@@ -961,25 +1006,639 @@ var _ = Describe("OSBuildEnvConfig Controller", func() {
 														})
 
 														Context("ConfigMap for the osbuild-worker config exists", func() {
+															const (
+																workerSSHKeysSecretName    = "osbuild-worker-ssh" // #nosec G101
+																workerSSHKeysPrivateKeyKey = "ssh-privatekey"
+																workerSSHKeysPublicKeyKey  = "ssh-publickey"
+															)
+
 															var (
 																workerOSBuildWorkerConfigConfigMap = corev1.ConfigMap{
 																	ObjectMeta: metav1.ObjectMeta{
 																		Namespace: operatorNamespace,
 																		Name:      workerOSBuildWorkerConfigConfigMapName,
 																	},
 																}
+
+																sshPrivateKey = []byte("sshPrivateKey")
+																sshPublicKey  = []byte("sshPublicKey")
+
+																workerSSHKeysSecret *corev1.Secret
 															)
 
 															BeforeEach(func() {
 																configMapRepository.EXPECT().Read(requestContext, workerOSBuildWorkerConfigConfigMapName, operatorNamespace).Return(&workerOSBuildWorkerConfigConfigMap, nil)
+																workerSSHKeysSecret = &corev1.Secret{
+																	ObjectMeta: metav1.ObjectMeta{
+																		Name:            workerSSHKeysSecretName,
+																		Namespace:       operatorNamespace,
+																		OwnerReferences: []metav1.OwnerReference{ownerReference},
+																	},
+																	Immutable: pointer.Bool(true),
+																	StringData: map[string]string{
+																		workerSSHKeysPrivateKeyKey: string(sshPrivateKey),
+																		workerSSHKeysPublicKeyKey:  string(sshPublicKey),
+																	},
+																}
 															})
 
-															It("Should return Done", func() {
+															It("Should return an error if failed to get the secret for the osbuild-worker ssh keys", func() {
+																// given
+																secretRepository.EXPECT().Read(requestContext, workerSSHKeysSecretName, operatorNamespace).Return(nil, errFailed)
 																// when
 																result, err := reconciler.Reconcile(requestContext, request)
 																// then
-																Expect(err).To(BeNil())
-																Expect(result).To(Equal(resultDone))
+																Expect(err).To(Equal(errFailed))
+																Expect(result).To(Equal(resultRequeue))
```

Comment:
I know it was merged but I think it needs to be changed (if not in that PR  in a new one) -  there is no reason to return a result like:   `return ctrl.Result{Requeue: true}, err` because as you can see here - https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/internal/controller/controller.go#L322 the error has been validated first

Reply:
Yes, you're right about the return value. We need to decide what's the right return value here. Should it return error at all? In any case, I don't think it should be part of this PR

Reply:
please open a ticket

Reply:
Done